File: backend/PvPet.Data/Entities/Pet.cs
using NetTopologySuite.Geometries;

namespace PvPet.Data.Entities;

public class Pet : Entity
{
    public string Name { get; set; } = default!;

    public int Variant { get; set; }

    public int Hp { get; set; }

    public int Food { get; set; }

    public int Gold { get; set; }

    public int Attack { get; set; }

    public int Armor { get; set; }

    public double AttackSpeed { get; set; }

    public int Crit { get; set; }

    public int CooldownSeconds { get; set; }

    public Point Location { get; set; } = default!;

    public int SecondsToRestockShop { get; set; }

    public Guid UserId { get; set; }

    public ICollection<PetItem> Items { get; set; } = null!;

    public ICollection<ShopItem> ShopItems { get; set; } = null!;

    public User User { get; set; } = default!;

    public ICollection<Fight> Fights { get; set; } = null!;
    public ICollection<PetFight> PetsFights { get; set; } = null!;
}

File: android/app/src/main/java/codebusters/pvpet/models/Pet.kt
package codebusters.pvpet.models

import com.google.android.gms.maps.model.LatLng

data class Pet(
    val id: String? = null,
    val name: String? = null,
    val variant: Int? = null,
    val hp: Int? = null,
    val food: Int? = null,
    val gold: Int? = null,
    val attack: Int? = null,
    val armor: Int? = null,
    val attackSpeed: Double? = null,
    val crit: Int? = null,
    val cooldownSeconds: Int? = null,
    val location: LatLng? = null,
    val secondsToRestockShop: Int? = null,
    val userId: String? = null,
    val items: List<PetItem>? = null,
    val shopItems: List<ShopItem>? = null,
    val user: List<User>? = null,
    val fights: List<PetFight>? = null,
)